src/components/date-picker/z-range-picker/index.tsx

```diff
@@ -0,0 +1,595 @@
+  handleClick() {
+    this.getFocusedInput();
+    this.addClassToInput();
+  }
+
+  @Listen("keydown", { target: "body", capture: true })
+  handleKeyDown(ev: KeyboardEvent) {
+    let currentInputState =
+      this.activeInput === RangePickerActiveInput.startInput
+        ? { picker: this.flatpickrInstance, index: 0 }
+        : { picker: this.flatpickrInstance2, index: 1 };
+
+    let isCalendarOpened = document.activeElement.closest(
+      ".flatpickr-calendar"
+    );
+    this.getFocusedInput();
+
+    if (ev.key === "Enter" || ev.key === " ") {
+      if (
+        document.activeElement.classList.contains(`${this.rangePickerId}`) ||
+        document.activeElement.classList.contains(`${this.rangePickerId}-2`)
+      ) {
+        !isCalendarOpened && currentInputState.picker.open();
+      }
+
+      let isPrevArrowEntered = document.activeElement.classList.contains(
+        "flatpickr-prev-month"
+      );
+      let isNextArrowEntered = document.activeElement.classList.contains(
+        "flatpickr-next-month"
+      );
+      let arrowPressed = isPrevArrowEntered || isNextArrowEntered;
+
+      arrowPressed && ev.key === " " && ev.preventDefault();
+
+      isPrevArrowEntered && currentInputState.picker.changeMonth(-1);
+      isNextArrowEntered && currentInputState.picker.changeMonth(1);
+    }
+  }
+
+  /** Set background color before or after first selected date, before selecting the second one */
+  @Listen("mouseover")
+  onMouseOver(e: CustomEvent) {
+    let currentInputState =
+      this.activeInput === RangePickerActiveInput.startInput
+        ? { picker: this.flatpickrInstance, index: 0 }
+        : { picker: this.flatpickrInstance2, index: 1 };
+
+    this.setRangeHoverStyle(
+      e,
+      currentInputState.picker,
+      this.element.getElementsByClassName("flatpickr-calendar")[
+        currentInputState.index
+      ]
+    );
+  }
+
+  componentDidLoad() {
+    let config = {
+      enableTime: this.mode === ZDatePickerMode.dateTime,
+      locale: Italian,
+      allowInput: true,
+      dateFormat:
+        this.mode === ZDatePickerMode.dateTime ? "d-m-Y - H:i" : "d-m-Y",
+      ariaDateFormat: "d F Y",
+      minuteIncrement: 1,
+      time_24hr: true,
+      onChange: (selectedDates, _dateStr, instance) => {
+        this.onDateSelect(selectedDates, instance);
+      },
+      onOpen: () => {
+        this.setAriaOptions();
+        this.setFlatpickrPosition();
+        this.setRangeStyle();
+        this.getFocusedInput();
+        this.getCurrentMonth();
+      },
+      onYearChange: () => {
+        this.setRangeStyle();
+      },
+      onMonthChange: () => {
+        this.setRangeStyle();
+      },
+      onKeyDown: () => {
+        this.setAriaOptions();
+      },
+      wrap: true,
+    };
+
+    this.flatpickrInstance = flatpickr(`.${this.rangePickerId}-container`, {
+      ...config,
+      mode: "multiple",
+      appendTo: this.element.children[0].children[0].children[0] as HTMLElement,
```

Comment:
sistemerei questi `appendTo`, magari usando `this.element.querySelector(`.${this.rangePickerId}-container`)` e con `-2` nell'altro